src/Agent.Sdk/Util/IOUtil.cs

```diff
@@ -223,6 +225,49 @@ public static void DeleteFile(string path)
             }
         }
 
+        public static  async Task DeleteDirectoryWithRetry(string path, CancellationToken cancellationToken, int retryCount = 3)
+        {
+            for (int i = 0; i < retryCount; i++)
+            {
+                try
+                {
+                    DeleteDirectory(path, cancellationToken);
+                }
+                // There is no reason to retry on DirectoryNotFoundException, SecruityException and UnauthorizedAccessException
+                catch (IOException)
+                {
+                    if (!cancellationToken.IsCancellationRequested || retryCount < 3)
+                    {
+                        // Pause execution briefly to allow the file to become accessible, but limit the wait to no more than 5 seconds
+                        await Task.Delay(Math.Max(retryCount, 5)*1000);
+                        continue;
+                    }
+                    throw;
+                }
+            }
+        }
+        public static async Task DeleteFileWithRetry(string path, CancellationToken cancellationToken, int retryCount = 3)
+        {
+            for (int i = 0; i < retryCount; i++)
+            {
```

Comment:
bug bug: looks like this will attempt to issue deletes up to the retry count even if successful?